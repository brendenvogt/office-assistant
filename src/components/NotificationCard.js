/**
 * @flow
 */
import React from 'react';
import {
  View,
  Text,
  ProgressBar,
  Card,
} from 'react-native-ui-lib';

import { colors } from '../styles';

type NotificationCardProps = {
  name: string,
  done: number,
  total: number,
  nextNotificationSeconds: number,
  totalIntervalSeconds: number,
};

export default function NotificationCard(props: NotificationCardProps) {
  const {
    name,
    done,
    total,
    nextNotificationSeconds,
  } = props;

  return (
    <Card borderRadius={3} marginH-10 marginT-15 flex>
      <View spread row padding-20>
        <View>
          <Text marginB-10 h1>{name}</Text>
          <Text>{done} / {total}</Text>
        </View>

        <View>
          <Text>{nextNotificationSeconds}</Text>
        </View>
      </View>
      <ProgressBar
        progress={done / total * 100}
        height={3}
        backgroundColor={colors.red}
        progressBackgroundColor={colors.lightenRed}
      />
    </Card>
  );
}
