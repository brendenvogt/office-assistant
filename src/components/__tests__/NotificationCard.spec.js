
/* eslint-disable no-undef */
import React from 'react';
import Enzyme from 'enzyme';

import NotificationCard from '../NotificationCard';

describe('Testing ToggleButton component', () => {
  it('renders as expected', () => {
    const wrapper = Enzyme.shallow(
      <NotificationCard
        name="Test Name"
        done={10}
        total={20}
        nextNotificationSeconds={300}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
