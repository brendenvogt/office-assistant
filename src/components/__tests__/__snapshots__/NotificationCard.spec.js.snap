// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ToggleButton component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationCard
    done={10}
    name="Test Name"
    nextNotificationSeconds={300}
    total={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "borderRadius": 3,
      "children": Array [
        <View
          padding-20={true}
          row={true}
          spread={true}
          useNativeDriver={true}
        >
          <View
            useNativeDriver={true}
          >
            <Text
              h1={true}
              marginB-10={true}
              useNativeDriver={true}
            >
              Test Name
            </Text>
            <Text
              useNativeDriver={true}
            >
              10
               / 
              20
            </Text>
          </View>
          <View
            useNativeDriver={true}
          >
            <Text
              useNativeDriver={true}
            >
              300
            </Text>
          </View>
        </View>,
        <ProgressBar
          backgroundColor="#e32d2f"
          height={3}
          progress={50}
          progressBackgroundColor="#FCECEC"
        />,
      ],
      "enableShadow": true,
      "flex": true,
      "marginH-10": true,
      "marginT-15": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              useNativeDriver={true}
            >
              <Text
                h1={true}
                marginB-10={true}
                useNativeDriver={true}
              >
                Test Name
              </Text>
              <Text
                useNativeDriver={true}
              >
                10
                 / 
                20
              </Text>
            </View>,
            <View
              useNativeDriver={true}
            >
              <Text
                useNativeDriver={true}
              >
                300
              </Text>
            </View>,
          ],
          "padding-20": true,
          "row": true,
          "spread": true,
          "useNativeDriver": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  h1={true}
                  marginB-10={true}
                  useNativeDriver={true}
                >
                  Test Name
                </Text>,
                <Text
                  useNativeDriver={true}
                >
                  10
                   / 
                  20
                </Text>,
              ],
              "useNativeDriver": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Test Name",
                  "h1": true,
                  "marginB-10": true,
                  "useNativeDriver": true,
                },
                "ref": null,
                "rendered": "Test Name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    10,
                    " / ",
                    20,
                  ],
                  "useNativeDriver": true,
                },
                "ref": null,
                "rendered": Array [
                  10,
                  " / ",
                  20,
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                useNativeDriver={true}
              >
                300
              </Text>,
              "useNativeDriver": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 300,
                "useNativeDriver": true,
              },
              "ref": null,
              "rendered": 300,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "#e32d2f",
          "height": 3,
          "progress": 50,
          "progressBackgroundColor": "#FCECEC",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "borderRadius": 3,
        "children": Array [
          <View
            padding-20={true}
            row={true}
            spread={true}
            useNativeDriver={true}
          >
            <View
              useNativeDriver={true}
            >
              <Text
                h1={true}
                marginB-10={true}
                useNativeDriver={true}
              >
                Test Name
              </Text>
              <Text
                useNativeDriver={true}
              >
                10
                 / 
                20
              </Text>
            </View>
            <View
              useNativeDriver={true}
            >
              <Text
                useNativeDriver={true}
              >
                300
              </Text>
            </View>
          </View>,
          <ProgressBar
            backgroundColor="#e32d2f"
            height={3}
            progress={50}
            progressBackgroundColor="#FCECEC"
          />,
        ],
        "enableShadow": true,
        "flex": true,
        "marginH-10": true,
        "marginT-15": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                useNativeDriver={true}
              >
                <Text
                  h1={true}
                  marginB-10={true}
                  useNativeDriver={true}
                >
                  Test Name
                </Text>
                <Text
                  useNativeDriver={true}
                >
                  10
                   / 
                  20
                </Text>
              </View>,
              <View
                useNativeDriver={true}
              >
                <Text
                  useNativeDriver={true}
                >
                  300
                </Text>
              </View>,
            ],
            "padding-20": true,
            "row": true,
            "spread": true,
            "useNativeDriver": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Text
                    h1={true}
                    marginB-10={true}
                    useNativeDriver={true}
                  >
                    Test Name
                  </Text>,
                  <Text
                    useNativeDriver={true}
                  >
                    10
                     / 
                    20
                  </Text>,
                ],
                "useNativeDriver": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Test Name",
                    "h1": true,
                    "marginB-10": true,
                    "useNativeDriver": true,
                  },
                  "ref": null,
                  "rendered": "Test Name",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      10,
                      " / ",
                      20,
                    ],
                    "useNativeDriver": true,
                  },
                  "ref": null,
                  "rendered": Array [
                    10,
                    " / ",
                    20,
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  useNativeDriver={true}
                >
                  300
                </Text>,
                "useNativeDriver": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": 300,
                  "useNativeDriver": true,
                },
                "ref": null,
                "rendered": 300,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#e32d2f",
            "height": 3,
            "progress": 50,
            "progressBackgroundColor": "#FCECEC",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
